code/iaas/logic/src/main/java/io/cattle/platform/process/instance/InstancePreCreate.java

```diff
@@ -69,6 +74,24 @@ public HandlerResult handle(ProcessState state, ProcessInstance process) {
         if (!data.isEmpty()) {
             return new HandlerResult(data);
         }
+
+        //Modifying ImageUuid as per the registry
+        String uuid = DataAccessor.fieldString(instance, InstanceConstants.FIELD_IMAGE_UUID);
```

Comment:
lets move all image related stuff to its separate method, something like setImage (similar to setNetworkMode(instance, labels, data); setDns(instance, labels, data))